include/boost/url/url.hpp

```diff
@@ -1292,7 +1293,27 @@ class BOOST_SYMBOL_VISIBLE url
     //
     //--------------------------------------------
 
-    /** Normalize everything.
+    /** Normalize the URL components
+
+        Applies Syntax-based normalization to the
+        URL.
+
+        @li The scheme is normalized to lowercase
+        @li The host is normalized to lowercase
+        @li Percent-encoding triplets are normalized
+        to uppercase letters
+        @li Percent-encoded octets that correspond
+        to unreserved characters are decoded
+        @li Redundant path-segments are removed
+
+        @par Exception Safety
+        Strong guarantee.
+        Calls to allocate may throw.
+
+        @par Specification
+        @li <a href="https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2"
+            >6.2.2 Syntax-Based Normalization (rfc3986)</a>
+
     */
     BOOST_URL_DECL
     url&
```

Comment:
I don't think `resolve` should be marked inline?

Reply:
Do you mean `resolve` or `normalize`? `normalize` is not `inline`, and this PR didn't touch the `resolve` function. Do you want me to remove `inline` from `resolve` and move it to the `.ipp` file in this PR?

Reply:
`resolve`. But its already in the .ipp file.

Reply:
The member function `resolve` is not `inline` and is defined in the `.ipp` file.

https://github.com/alandefreitas/url/blob/11c76e7395821e254d4bd0fa0edc8ccd0ea38a43/include/boost/url/url.hpp#L1366-L1372

https://github.com/alandefreitas/url/blob/11c76e7395821e254d4bd0fa0edc8ccd0ea38a43/include/boost/url/impl/url.ipp#L1827-L1833

The `friend` function `resolve` is `inline` and defined in the `.hpp` file:

https://github.com/alandefreitas/url/blob/11c76e7395821e254d4bd0fa0edc8ccd0ea38a43/include/boost/url/url.hpp#L1328-L1335

https://github.com/alandefreitas/url/blob/11c76e7395821e254d4bd0fa0edc8ccd0ea38a43/include/boost/url/url.hpp#L1415-L1426

So do you want me to change the `friend` function?